app/src/main/java/com/example/compassofukraine/viewModel/DetailedEventViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.compassofukraine.viewModel
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.model.DetailedEvent
+import com.example.useCase.AddEventToFavoriteUseCase
+import com.example.useCase.GetDetailsEventsUseCase
+import com.example.useCase.IsEventInFavoriteUseCase
+import com.example.useCase.RemoveEventFromFavoriteUseCase
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import java.util.concurrent.TimeoutException
+
+class DetailedEventViewModel(
+    private val getDetailsEventsUseCase: GetDetailsEventsUseCase,
+    private val addEventToFavoriteUseCase: AddEventToFavoriteUseCase,
+    private val removeEventFromFavoriteUseCase: RemoveEventFromFavoriteUseCase,
+    private val isEventInFavoriteUseCase: IsEventInFavoriteUseCase
+) : ViewModel() {
+    private val _event: MutableState<DetailedEvent?> = mutableStateOf(null)
```

Comment:
Instead of using nullable DetailedEvent we should wrap this variable with sealed class Result whose should contain loading, success and error states